src/v/cloud_storage/remote_partition.cc

```diff
@@ -53,26 +53,31 @@ namespace cloud_storage {
 using data_t = model::record_batch_reader::data_t;
 using storage_t = model::record_batch_reader::storage_t;
 
-remote_partition::iterator remote_partition::materialize_segment(
+remote_partition::iterator remote_partition::get_or_materialize_segment(
   const remote_segment_path& path,
   const segment_meta& meta,
   segment_units unit) {
     _as.check();
-    auto base_kafka_offset = meta.base_offset - meta.delta_offset;
     auto st = std::make_unique<materialized_segment_state>(
       meta, path, *this, std::move(unit));
     auto [iter, ok] = _segments.insert(
       std::make_pair(meta.base_offset, std::move(st)));
-    vassert(
-      ok,
-      "Segment with base log offset {} and base kafka offset {} is already "
-      "materialized, max offset of the new segment {}, max offset of the "
-      "existing segment {}",
-      meta.base_offset,
-      base_kafka_offset,
-      meta.committed_offset,
-      iter->second->segment->get_max_rp_offset());
-    _ts_probe.segment_materialized();
+
+    if (ok) {
+        vlog(
+          _ctxlog.trace,
```

Comment:
why not `debug`?

Reply:
I expected that to be more frequent, but now that I think about it again it doesn't really make sense.

Reply:
Changed.